packages/mgt-components/src/components/mgt-teams-channel-picker/mgt-teams-channel-picker.ts

```diff
@@ -880,30 +894,77 @@ export class MgtTeamsChannelPicker extends MgtTemplatedComponent {
     }
 
     this._isDropdownVisible = true;
+    this.toggleChevron();
+    this.resetFocusState();
   }
 
   private lostFocus() {
-    this._isFocused = false;
     const input = this._input;
     if (input) {
-      input.textContent = this._inputValue = '';
+      input.value = this._inputValue = '';
+      input.textContent = '';
     }
 
+    this._isFocused = false;
     this._isDropdownVisible = false;
     this.filterList();
+    this.toggleChevron();
+    this.requestUpdate();
+
+    if (this._selectedItemState !== undefined) {
+      this.showCloseIcon();
+    }
   }
 
   private selectChannel(item: ChannelPickerItemState) {
-    if (this._selectedItemState !== item) {
-      this._selectedItemState = item;
-      this.fireCustomEvent('selectionChanged', this.selectedItem);
+    if (item && this._selectedItemState !== item) {
+      this._input.setAttribute('disabled', 'true');
+    } else {
+      this._input.removeAttribute('disabled');
     }
+    this._selectedItemState = item;
+    this.lostFocus();
+    this.fireCustomEvent('selectionChanged', item ? [this.selectedItem] : []);
```

Comment:
This doesn't matcht the typing in the comments and doesn't much make sense. We can only have a single channel selected in this control so it should be a singular channel or null on the detail depending on if we selected an item or removed the selected item